sqlmesh/core/model/definition.py

```diff
@@ -2332,7 +2332,7 @@ def _list_of_calls_to_exp(value: t.List[t.Tuple[str, t.Dict[str, t.Any]]]) -> ex
 
 def _extract_audit_expressions(
     audits: t.Optional[dict[str, Audit]] = None,
-    inline_audits: t.Optional[dict[str, Audit]] = None,
```

Comment:
why does your change make it so this one has to change? this seems like a regression

Reply:
Yeah I understand it may seem like that at first, but it has to do with Python's type system. The reason this needs to be changed is that Python `dict`s are _invariant_ in their value (and key) types. There's a good explanation as to why that is [here](https://stackoverflow.com/questions/78537075/why-is-dictint-int-incompatible-with-dictint-int-str/78537409#78537409). This means you can't pass a `dict[str, ModelAudit]` as a `dict[str, Audit]` argument. However, `Mapping` is a superclass of `dict` and is _covariant_ in value types. Therefore, changing this type to `Mapping[str, Audit]` satisfies the type constraints (and `inline_audits` isn't mutated at all in the function body, so we don't need a mutable type here).